internal/terraform/json_test.go

```diff
@@ -135,28 +135,24 @@ func TestResourceProviderConfig(t *testing.T) {
 	p := unmarshalTestPlan(t)
 
 	tests := []struct {
-		kind    string
-		name    string
-		wantErr bool
+		kind string
+		name string
 	}{
-		// Empty address - should return err.
-		{"", "", true},
+		// Empty address - should return empty.
+		{"", ""},
 
-		// No provider for this resource - should return err.
-		{"google_storage_bucket", "some_other_bucket", true},
+		// No provider for this resource - should return empty.
+		{"google_storage_bucket", "some_other_bucket"},
 
-		// Provider config exists - should not return err.
-		{"google_storage_bucket", "gcs_tf_bucket", false},
+		// Provider config exists - should not return empty.
+		{"google_storage_bucket", "gcs_tf_bucket"},
 	}
 	for _, testcase := range tests {
-		config, err := resourceProviderConfig(testcase.kind, testcase.name, p)
-		if testcase.wantErr {
-			if err == nil {
-				t.Errorf("resourceProviderConfig(%q, %q, %v) returned nil err; wanted err != nil", testcase.kind, testcase.name, testPlanPath)
-			}
-		} else {
-			if cmp.Equal(config, ProviderConfig{}, cmpopts.EquateEmpty()) {
-				t.Errorf("resourceProviderConfig(%q, %q, %v) returned empty provider config; want nonempty config", testcase.kind, testcase.name, testPlanPath)
+		config, ok := resourceProviderConfig(testcase.kind, testcase.name, p)
+
+		if !ok {
+			if diff := cmp.Diff(config, ProviderConfig{}, cmpopts.EquateEmpty()); diff != "" {
```

Comment:
This wasn't exactly what I meant :P but it's fine to leave this in.

The more important case is when it is "ok" (i.e. the provider config exists). I thought you may want to check the provider config against what you expect to get.

Reply:
I've changed it now to check against a specific expected provider config. I was kind of hoping to avoid doing that since it's somewhat complicated to get the types right, but I realized that I could just copy the output of cmp.Diff and use that as the expected output.